sesame2spiner/generate_files.cpp

```diff
@@ -273,34 +274,110 @@ void getMatBounds(int i, int matid, const SesameMetadata &metadata, const Params
               << "shrinkleBounds > 0 and siemin or siemax set" << std::endl;
   }
 
-  int ppdRho = params.Get("numrho/decade", PPD_DEFAULT);
-  int numRhoDefault = getNumPointsFromPPD(rhoMin, rhoMax, ppdRho);
+  int ppdRho = params.Get("numrho/decade", PPD_DEFAULT_RHO);
+  int numRhoDefault = Bounds::getNumPointsFromPPD(rhoMin, rhoMax, ppdRho);
 
-  int ppdT = params.Get("numT/decade", PPD_DEFAULT);
-  int numTDefault = getNumPointsFromPPD(TMin, TMax, ppdT);
+  int ppdT = params.Get("numT/decade", PPD_DEFAULT_T);
+  int numTDefault = Bounds::getNumPointsFromPPD(TMin, TMax, ppdT);
 
-  int ppdSie = params.Get("numSie/decade", PPD_DEFAULT);
-  int numSieDefault = getNumPointsFromPPD(sieMin, sieMax, ppdSie);
+  int ppdSie = params.Get("numSie/decade", PPD_DEFAULT_T);
+  int numSieDefault = Bounds::getNumPointsFromPPD(sieMin, sieMax, ppdSie);
 
   int numRho = params.Get("numrho", numRhoDefault);
   int numT = params.Get("numT", numTDefault);
   int numSie = params.Get("numsie", numSieDefault);
 
-  Real rhoAnchor = metadata.normalDensity;
-  Real TAnchor = 298.15;
+  const Real TAnchor = 298.15;
+  Real rhoAnchor = params.Get("rho_fine_center", metadata.normalDensity);
+  if (std::isnan(rhoAnchor) || rhoAnchor <= 0 || rhoAnchor > 1e8) {
+    std::cerr << "WARNING [" << matid << "] "
+              << "normal density ill defined. Setting it to a sensible default."
+              << std::endl;
+    rhoAnchor = 1;
+  }
+
+  // Piecewise grids stuff
+  bool piecewiseRho = params.Get("piecewiseRho", true);
+  bool piecewiseT = params.Get("piecewiseT", true);
+  bool piecewiseSie = params.Get("piecewiseSie", true);
+
+  Real ppd_factor_rho_lo = params.Get("rhoCoarseFactorLo", COARSE_FACTOR_DEFAULT_RHO_LO);
+  Real ppd_factor_rho_hi = params.Get("rhoCoarseFactorHi", COARSE_FACTOR_DEFAULT_RHO_HI);
+  Real ppd_factor_T = params.Get("TCoarseFactor", COARSE_FACTOR_DEFAULT_T);
+  Real ppd_factor_sie = params.Get("sieCoarseFactor", COARSE_FACTOR_DEFAULT_T);
+  Real rho_fine_diameter =
+      params.Get("rhoFineDiameterDecades", RHO_FINE_DIAMETER_DEFAULT);
+  Real TSplitPoint = params.Get("TSplitPoint", T_SPLIT_POINT_DEFAULT);
+
+  Real rho_fine_center = rhoAnchor;
```

Comment:
I might be missing changes below, but it looks like all of these could be `const`.

There are other reasons to advocate for lots of `const`, but I think from a readability perspective it's helpful to understand which variables may change and which are never going to change.

Reply:
Agreed. I'll add const tags.